scripts/nodes/setup-ros-node.bash

```diff
@@ -0,0 +1,340 @@
+NODE_YAML="config/${FILE_NAME}.yaml"
+NODE_LAUNCH="launch/${FILE_NAME}.launch.xml"
+
+VC_H="include/$PKG_NAME/visibility_control.h"
+NODE_HPP="include/${PKG_NAME}/${FILE_NAME}.hpp"
+NODE_PARAMS_YAML="src/${FILE_NAME}.yaml"
+NODE_CPP="src/${FILE_NAME}.cpp"
+
+# Copy files
+if [[ ! -f "$VC_H" ]]; then
+  cp -n "$NODE_TEMPLATES/dummy_package_namespace/visibility_control.h" $VC_H
+fi
+cp -n "$NODE_TEMPLATES/dummy_node_params.yaml" ${NODE_YAML}
+cp -n "$NODE_TEMPLATES/dummy_node.launch.xml" "${NODE_LAUNCH}"
+cp -n "$NODE_TEMPLATES/dummy_package_namespace/dummy_node.hpp" "${NODE_HPP}"
+cp -n "$NODE_TEMPLATES/dummy_node.yaml" "${NODE_PARAMS_YAML}"
+cp -n "$NODE_TEMPLATES/dummy_node.cpp" "${NODE_CPP}"
+
+echo -e "${TERMINAL_COLOR_USER_NOTICE}Template files copied.${TERMINAL_COLOR_NC}"
+
+# Add license header to the files
+FILES_TO_LICENSE=("$NODE_HPP" "$NODE_CPP")
+if [[ "$package_configured" == "no" ]]; then
+  FILES_TO_LICENSE+=("$VC_H")
+fi
+TMP_FILE=".f_tmp"
+if [[ "$LICENSE_HEADER" != "" ]]; then
+  touch $TMP_FILE
+  for FILE_TO_LIC in "${FILES_TO_LICENSE[@]}"; do
+    cat $LICENSE_HEADER > $TMP_FILE
+    sed "1,13d" $FILE_TO_LIC >> $TMP_FILE # delete first 13 lines which correspond to fake license
+    mv $TMP_FILE $FILE_TO_LIC
+    sed -i "s/\\\$YEAR\\\$/${YEAR_ON_LICENSE}/g" $FILE_TO_LIC
+    sed -i "s/\\\$NAME_ON_LICENSE\\\$/${NAME_ON_LICENSE}/g" $FILE_TO_LIC
+  done
+#   echo "Licence header added to files: ("`declare -p FILES_TO_LICENSE`")"
+fi
+
+# sed all needed files
+FILES_TO_SED=("${FILES_TO_LICENSE[@]}")
+FILES_TO_SED+=("$NODE_YAML" "$NODE_LAUNCH" "$NODE_PARAMS_YAML")
+
+for SED_FILE in "${FILES_TO_SED[@]}"; do
+  sed -i "s/\\\$PKG_NAME\\\$/${PKG_NAME}/g" $SED_FILE
+  sed -i "s/\\\$RUNTIME_CONFIG_PKG_NAME\\\$/${PKG_NAME}/g" $SED_FILE
+  sed -i "s/\\\$FILE_NAME\\\$/${FILE_NAME}/g" $SED_FILE
+
+  sed -i "s/TEMPLATES__NODES__DUMMY_PACKAGE_NAMESPACE/${PKG_NAME^^}/g" $SED_FILE # package name for include guard
+  sed -i "s/TEMPLATES__NODES__VISIBILITY/${PKG_NAME^^}__VISIBILITY/g" $SED_FILE # visibility defines
+  sed -i "s/dummy_package_namespace/${PKG_NAME}/g" $SED_FILE # package name for includes
+  sed -i "s/dummy_node/${FILE_NAME}/g" $SED_FILE # file name
+  sed -i "s/DUMMY_NODE/${FILE_NAME^^}/g" $SED_FILE # file name for include guard
+  sed -i "s/DummyClassName/${CLASS_NAME}/g" $SED_FILE # class name
+  sed -i "s/DummyNode/${CLASS_NAME}/g" $SED_FILE # class name
+done
+
+# CMakeLists.txt: Remove comments if there any and add library
+DEL_STRINGS=("# uncomment the following" "# further" "# find_package(<dependency>")
+
+for DEL_STR in "${DEL_STRINGS[@]}"; do
+  sed -i "/$DEL_STR/d" CMakeLists.txt
+done
+
+TMP_FILE=".f_tmp"
+touch $TMP_FILE
+
+# manage dependencies here
+if [[ "$package_configured" == "no" ]]; then
+
+  # Add `THIS_PACKAGE_INCLUDE_DEPENDS` structure to simplify the rest of the file
+  TEST_LINE=`awk '$1 == "find_package(ament_cmake" { print NR }' CMakeLists.txt`  # get line before `ament_cmake` dependency
```

Comment:
`head` and `tail` commands are having trouble parsing the expanded string, I'm getting these warnings:

```
tail: invalid number of lines: ‘+’
head: unrecognized option '--1'
Try 'head --help' for more information.
tail: invalid number of lines: ‘+’
tail: invalid number of lines: ‘+’
tail: invalid number of lines: ‘+’
```
This is happening both when `$package_configured` is `no` or `yes`

Is this working as expected on your machine?

Reply:
What is the initial state of your package when testing this?